### DISCUSSION TITLE
Controlling an auxiliary axis from a plugin

### DISCUSSION THREAD

**Discussion by lost-tty** [grblHAL/core#613] [General]
I am in the process of integrating grblHAL-core into an existing CNC machine. This machine is using an auxiliary stepper for the automatic tool changer to fasten or loosen clamp. This is basically an axis with a homing switch and it is connected to the MCU just like the primary axes (step and direction pin).

Could you give me advice on how to use this axis from within a plugin without making it available to general G-Codes programs?

**Comment by phil-barrett** [comment]
There is a tool change discussion [here](https://github.com/grblHAL/core/discussions/577).  Making the axis useable via GCode and macros seems like the easier way to go.

**Comment by terjeio** [comment]
> Could you give me advice on how to use this axis from within a plugin without making it available to general G-Codes programs?

For now it is not possible unless you add code to drive it in the plugin itself. This may change in a later version though.
